Libraries/LibWeb/HTML/HTMLFormElement.cpp

```diff
@@ -234,31 +235,42 @@ WebIDL::ExceptionOr<void> HTMLFormElement::submit_form(GC::Ref<HTMLElement> subm
     // 17. Let enctype be the submitter element's enctype.
     auto encoding_type = encoding_type_state_from_form_element(submitter);
 
-    // 18. Let target be the submitter element's formtarget attribute value, if the element is a submit button and has
-    //     such an attribute. Otherwise, let it be the result of getting an element's target given submitter's form
-    //     owner.
-    auto target = submitter->attribute(AttributeNames::formtarget).value_or(get_an_elements_target());
+    // 18. Let formTarget be null.
+    Optional<String> form_target;
+
+    // 19. If the submitter element is a submit button and it has a formtarget attribute, then set formTarget to the
+    //     formtarget attribute value.
+    if (is<FormAssociatedElement>(*submitter) && dynamic_cast<FormAssociatedElement*>(submitter.ptr())->is_submit_button()) {
```

Comment:
this does the same dynamic cast twice.

`if (auto* submit_button = dynamic_cast<FormAssociatedElement*>(submitter.ptr()); submit_button && submit_button->is_submit_button())` ?

Reply:
Thanks! This was a consequence of me trying to use `as_if()` and the compiler rejecting it. Looks like #3414 will solve that so I'll hang back until that's merged.